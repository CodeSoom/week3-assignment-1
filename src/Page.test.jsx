import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Page from './Page';

test('Page 컴포넌트 구성요소는 h1, Input, List로 이루어져있다.', () => {
  const { container, getByPlaceholderText } = render(<Page tasks={[]} />);

  expect(container).toContainHTML('h1');
  expect(container).toHaveTextContent('To-do');

  // 여기를 Input, List로 테스트 할 수 있는 방법은 없는 것인가요?
  expect(container).toContainHTML('input');
  getByPlaceholderText('할 일을 입력해 주세요');

  expect(container).toContainHTML('ol');
  expect(container).toHaveTextContent('추가');

  expect(container.firstChild).toMatchSnapshot();
});

test('할 일을 작성하면, list에 목록이 추가된다.', () => {
  const onChange = jest.fn();
  const onClick = jest.fn();

  const { container, getByPlaceholderText, getByText } = render(
    <Page tasks={[]} onChangeTitle={onChange} onClickAddTask={onClick} />,
  );
  const input = getByPlaceholderText('할 일을 입력해 주세요');
  const submitButton = getByText('추가');

  userEvent.type(input, '오늘의 할 일');
  expect(onChange).toBeCalled();

  userEvent.click(submitButton);
  expect(onClick).toBeCalled();

  // 추가가 되는 것을 테스트할 수 없음..
  // expect(container).toHaveTextContent('오늘의 할 일');
  // expect(container).toContainHTML('')
});

test('할 일을 완료하면, 리스트 목록에서 제거된다.', () => {
  const tasks = [{
    id: 1,
    title: '뭐라도 하기',
  }];

  const onDeleteItem = jest.fn();

  const { container, getByText } = render(
    <Page tasks={tasks} onClickDeleteTask={onDeleteItem} />,
  );

  expect(container).toHaveTextContent('뭐라도 하기');
  expect(container).toHaveTextContent('완료');
  expect(onDeleteItem).not.toBeCalled();

  userEvent.click(getByText('완료'));
  expect(onDeleteItem).toBeCalledWith(1);

  // 삭제가 되는 것을 테스트할 수 없음..
  expect(container).not.toHaveTextContent('뭐라도 하기');
});
