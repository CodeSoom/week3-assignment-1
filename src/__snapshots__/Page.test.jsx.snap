// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page 컴포넌트 구성요소는 h1, Input, List로 이루어져있다. 1`] = `
<div>
  <h1>
    To-do
  </h1>
  <p>
    <label
      for="input-task-title"
    >
      할 일
    </label>
    <input
      id="input-task-title"
      placeholder="할 일을 입력해 주세요"
      type="text"
      value=""
    />
    <button
      type="button"
    >
      추가
    </button>
  </p>
  <p>
    할 일이 없어요!
  </p>
</div>
`;
